app/controllers/frontend_log_controller.rb

```diff
@@ -77,4 +77,15 @@ def valid_event?
   def valid_payload?
     !log_params[:payload].nil?
   end
+
+  def hash_from_method_kwargs(hash, method)
+    method_kwargs(method).index_with { |key| hash[key.to_s] }
+  end
```

Comment:
in hindsight, this starts to feel like the kind of work we'd want to extract from a controller maybe? and move to like a "analytics from params" class but that was like this long before this PR so no need to change here, just thinking out loud

Reply:
I'm always open to these sorts of "doing it the right way" suggestions, so I'll look into refactoring that.

Reply:
Does 03e378d5a look like what you had in mind? Open to any suggestions.

Reply:
Yeah! I was even thinking the `bind_call` stuff too might be a good fit to extract too, so the controller would be

```
def create
  FrontEndLogger.new(analytics).log_event(events: EVENT_MAP, params: params)
end
```

Reply:
> I was even thinking the `bind_call` stuff too might be a good fit to extract too

Yeah, I could see that being useful, but I think there's some desire (that I share) to try to move away from some of the overly-permissive defaulting that currently happens with this logging, which I'd imagine could have some impact on how we might want to implement that, and is a bit bigger than I'd want to bite off here.

Reply:
ok! another intermediate idea, but also happy to accept what we have here and continue to noodle/iterate

```ruby
def create
  did_match_event = FrontEndLogger.new(analytics).log_event(events: EVENT_MAP, params: log_params)

  if !did_match_event
    # free-form logging here
  end

  render json: { success: true }, status: :ok
end
```

Reply:
I just went ahead and moved it all. How's it looking after 24cb8ecad ?